### PR TITLE
Change OpenAPI encoding of heterogeneous arrays

### LINKED ISSUE #856: Encoding of tuples in OpenAPI does not conform to spec
Example; this definition
```scala
  val foo: Endpoint[Unit, (String, Long)] =
    endpoint(
      request = get(path / "foo"),
      response = ok(jsonResponse[(String, Long)])
    )
```
produces this OpenAPI definition for response content (`items` is an array of SchemaObject)
```json
"content": {
  "application/json": {
    "schema": {
      "type": "array",
      "items": [
        {
          "type": "string"
        },
        {
          "type": "integer",
          "format": "int64"
        }
      ]
    }
  }
}
```
but, OpenAPI spec is explicitly against that. In [this part](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#schema-object), it's stated:
> items - Value MUST be an object and not an array. Inline or referenced schema MUST be of a Schema Object and not a standard JSON Schema. items MUST be present if the type is array.

OpenAPI 3.0 [doesn't seem to have a proper encoding for tuples](https://stackoverflow.com/a/57465199), unfortunately, but it might be better to produce (not perfectly precise) approximation that conforms spec.

### PR DESCRIPTION
**Pull request by minedeljkovic**
Fixes #856

### CODE REVIEW

File: openapi/openapi/src/main/scala/endpoints4s/openapi/model/OpenApi.scala [hunk 1]
```diff
@@ -89,14 +89,27 @@ object OpenApi {
...
+            fields += "items" -> schemaJson(value)
+          case Right(value) =>
+            /** Best effort to represent the heterogeneous array in OpenAPI 3.0
+              * This should be changed with OpenAPI 3.1 and more idiomatic representation using `prefixItems`
+              */
+            fields ++= List(
+              "items" -> schemaJson(
+                Schema.OneOf(
+                  alternatives = Schema.EnumeratedAlternatives(value),
+                  description = None,
```
Comment (minedeljkovic) [inline openapi/openapi/src/main/scala/endpoints4s/openapi/model/OpenApi.scala:104]: I was thinking of maybe using the description here to stress additional constraint (fixed positions of elements in a tuple), that cannot be enforced by OpenAPI 3.0 features (along the lines of [this example](https://stackoverflow.com/a/57465199/15821235)), but I'm not sure what would be an appropriate format/language.

File: openapi/openapi/src/main/scala/endpoints4s/openapi/model/OpenApi.scala [hunk 2]
```diff
@@ -89,14 +89,26 @@ object OpenApi {
...
+            fields += "items" -> schemaJson(value)
+          case Right(value) =>
+            // Best effort (not 100% accurate) to represent the heterogeneous array in OpenAPI 3.0
+            // This should be changed with OpenAPI 3.1 and more idiomatic representation using `prefixItems`
+            fields ++= List(
+              "items" -> schemaJson(
+                Schema.OneOf(
+                  alternatives = Schema.EnumeratedAlternatives(value),
+                  description = None,
+                  example = None,
```
Comment (julienrf) [inline openapi/openapi/src/main/scala/endpoints4s/openapi/model/OpenApi.scala:104]: We could take the first defined example of the tuple elements, if any.

File: openapi/openapi/src/main/scala/endpoints4s/openapi/model/OpenApi.scala (hunk 1 above)
Comment (julienrf) [reply openapi/openapi/src/main/scala/endpoints4s/openapi/model/OpenApi.scala:103]: I think it’s fine to leave it empty. What do others think?

**Review by julienrf — APPROVED** [review]
Thank you @minedeljkovic!

(I’ve just edited my review and deleted a wrong comment.)

**Pull request merged**